### PR TITLE
ACQ-751: Company Name component now support to be optional.

### PR DESCRIPTION
**Pull request by maguero**
### Description
Make Company Name component optional or required by parameter.

### Ticket
https://financialtimes.atlassian.net/browse/ACQ-751

### Screenshots
Screenshot 2021-03-01 at 12 36 39

### CODE REVIEW

File: components/company-name.spec.js [hunk 1]
```diff
@@ -83,4 +83,20 @@ describe('CompanyName', () => {
 			'Organisation name'
 		);
 	});
+
+	it('renders as required flield', () => {
```
Comment (ludrob) [inline components/company-name.spec.js:87]:
```suggestion
	it('renders as required field', () => {
```

File: components/company-name.spec.js [hunk 2]
```diff
@@ -83,4 +83,20 @@ describe('CompanyName', () => {
...
+		const component = mount(CompanyName({}));
+
+		expect(component.find('#companyNameField').hasClass('o-forms-field--optional')).toBe(false);
+		expect(component.find('input#companyName').prop('required')).toBe(true);
+	});
+
+	it('renders as optional flield', () => {
```
Comment (ludrob) [inline components/company-name.spec.js:94]:
```suggestion
	it('renders as optional field', () => {
```

File: components/company-name.jsx [hunk 3]
```diff
@@ -11,7 +11,13 @@ export function CompanyName({
 	isDisabled = false,
 	fieldLabel = 'Company name',
 	placeholder = 'Please enter your company name',
+	isRequired = true,
 }) {
+	const divClassNames = classNames([
```
Comment (ekocian) [inline components/company-name.jsx:16]: **question:**  when isRequired is `true` divClassNames will result on `o-forms-field o-forms-field--optional`. Is that duplication desired?

Comment (maguero) [reply components/company-name.jsx:16]: good question! I just took other components implementation as example, but I didn't check if both are needed.

**Review by ekocian — APPROVED** [review]
LGTM!

**Pull request merged**